### ISSUE TITLE
docker-in-docker fails with the `ubuntu:latest` image

### ISSUE THREAD

**Issue by samruddhikhandale** [actions/runner-images#6980]
### Description

Reported in https://github.com/devcontainers/ci/issues/192

Docker fails with `Is the docker daemon running` error & the when the container is started, we can see `sed: couldn't flush stdout: Device or resource busy` error.

These errors are flakey (action fails ~20% of times) and only happens in a GitHub action with `runs-on: ubuntu-latest` in the workflow. 

Note: The errors does not appear with `runs-on: ubuntu-20.04`

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

https://github.com/customink/dnd-demo/actions/runs/3990966349/jobs/6845376920

### Is it regression?

Not sure

### Expected behavior

Docker should always work in a `ubuntu:latest` action runner.

### Actual behavior

Docker fails to start.

### Repro steps

Reference repo - https://github.com/customink/dnd-demo

**Comment by Alexey-Ayupov** [comment]
Hello @samruddhikhandale, we'll take a look at it.

**Comment by Alexey-Ayupov** [comment]
The user "runner" used on runner-images is a member of a "docker" group, so you shouldn't expect such problems. 
However to understand the nature of the problem, could you please run the docker-in-docker task without using "devcontainers/ci@v0.2" action?
We would like to make sure that the root cause is not the action itself.

**Comment by metaskills** [comment]
So @samruddhikhandale asked if I would you be interested to test this hypothesis?. I would, but I'm not sure what "run the docker-in-docker task" would entail. From previous tests (https://github.com/customink/dnd-demo/actions/runs/3981703221) we know that devcontainers/ci has docker working in the runner. If you look at the good/bad jobs you can see it fails to start the container from the CLI due to some sed error. 

Basically, I'm not sure how to exercise a test for y'all outside of running the devcontainer/ci as intended. Do y'all have recommendations?

**Comment by Chuxel** [comment]
@metaskills One way to do this could just be to use the docker CLI directly with the universal image given it has DinD already set up.

`docker run --privileged -it --rm mcr.microsoft.com/devcontainers/universal bash -c 'docker ps'`

The `devcontainer` CLI could also be used - though that's what the Action ultimately uses.

**Comment by mikhailkoliada** [comment]
@Chuxel `-i` would not work here as action image is being run with non-interactive shell

@samruddhikhandale @metaskills 

```csl
 docker run -d --name=dind -t -v /var/run/docker.sock:/var/run/docker.sock docker
 docker exec -t dind docker ps
```

These commands is the bare minimum to indicate that dind in the images is functional as the concept, i.e. as you can see it does not reveal socket-specific problems described above.

**Comment by samruddhikhandale** [comment]
Hi 👋 

@Alexey-Ayupov Wondering if you have an update, we have another user reporting the same issue in https://github.com/devcontainers/features/issues/372 Thanks!

**Comment by metaskills** [comment]
@mikhailkoliada Thanks for those commands. They fail at the same percentage on ubuntu-latest as the full docker in docker stuff. Does that help provide a test case for resolution?

https://github.com/customink/dnd-demo/actions/runs/4089367131

**Comment by mikhailkoliada** [comment]
Hello! We are still unable to reproduce the failure. Could you please track the `systemctl status docker` call if failure happens?

**Comment by mikhailkoliada** [comment]
Please reopen the ticket once you have more data for us we could analyze, currently we have not enough information.

**Comment by metaskills** [comment]
I'm confused. I have a repo (link above) that consistently demonstrates the issue in a reproducible way. Whose responsibility is it to track the `systemctl status docker`? Are you saying that this is a devcontainer/cli issue?

**Comment by metaskills** [comment]
I did another test today and did not see a failure in 20 runs. So maybe this has been fixed. https://github.com/customink/dnd-demo/actions/runs/5135465641
